druid-shell/src/platform/x11/window.rs

```diff
@@ -1212,15 +1260,19 @@ impl PresentData {
         pixmap: Pixmap,
         window_id: u32,
         region: &Region,
+        scale: Scale,
     ) -> Result<(), Error> {
         let x_rects: Vec<Rectangle> = region
             .rects()
             .iter()
-            .map(|r| Rectangle {
-                x: r.x0 as i16,
-                y: r.y0 as i16,
-                width: r.width() as u16,
-                height: r.height() as u16,
+            .map(|r| {
+                let r = r.to_px(scale);
+                Rectangle {
+                    x: r.x0 as i16,
+                    y: r.y0 as i16,
```

Comment:
I think you'll want to round these. `r.px(scale)` should essentially have integer coordinates, but I think it's possible for the coordinates to be just slightly smaller than an integer. In that case, we should round up, but this will round down.